File: tutorials/30-days-of-code/30-scope.py
# Tutorials > 30 Days of Code > Day 14: Scope
# Learn about the scope of an identifier.
#
# https://www.hackerrank.com/challenges/30-scope/problem
#

class Difference:
    def __init__(self, a):
        self.__elements = a
# (skeliton_head) ----------------------------------------------------------------------

    def __init__(self, a):
        self.a = a

    def computeDifference(self):
        self.maximumDifference = max(self.a) - min(self.a)

# (skeliton_tail) ----------------------------------------------------------------------
# End of Difference class

_ = input()
a = [int(e) for e in input().split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)

File: tutorials/30-days-of-code/30-scope.cpp
// Tutorials > 30 Days of Code > Day 14: Scope
// Learn about the scope of an identifier.
//
// https://www.hackerrank.com/challenges/30-scope/problem
//

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Difference {
    private:
    vector<int> elements;

  	public:
  	int maximumDifference;
// (skeliton_head) ----------------------------------------------------------------------

    Difference(const vector<int>& a)
    {
        maximumDifference = *std::max_element(a.begin(), a.end()) - *std::min_element(a.begin(), a.end());
    }

    void computeDifference() const
    {}

// (skeliton_tail) ----------------------------------------------------------------------
}; // End of Difference class

int main() {
    int N;
    cin >> N;

    vector<int> a;

    for (int i = 0; i < N; i++) {
        int e;
        cin >> e;

        a.push_back(e);
    }

    Difference d(a);

    d.computeDifference();

    cout << d.maximumDifference;

    return 0;
}
